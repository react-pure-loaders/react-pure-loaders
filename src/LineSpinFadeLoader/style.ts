import { keyframes } from '@emotion/react';
import { globalAnimation, lines } from '../mixins';
import delay from '../utilities';
import { Grid } from '../types';

const radius = '20px';

const lineSpinFade = keyframes({
  '100%': {
    opacity: 1,
  },
  '50%': {
    opacity: 0.3,
  },
});

function lineSpinFadeLoader() {
  const n = 8;
  let i = 1;
  const grid: Grid = {};
  for (; i < 9; i += 1) {
    const quarter = (parseInt(radius, 10) / 2) + (parseInt(radius, 10) / 5.5);

    if (i === 1) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: 0,
        top: radius,
      };
    } else if (i === 2) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: quarter,
        top: quarter,
        transform: 'rotate(-45deg)',
      };
    } else if (i === 3) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: radius,
        top: 0,
        transform: 'rotate(90deg)',
      };
    } else if (i === 4) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: quarter,
        top: `-${quarter}`,
        transform: 'rotate(45deg)',
      };
    } else if (i === 5) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: '0',
        top: `-${radius}`,
      };
    } else if (i === 6) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: `-${quarter}`,
        top: `-${quarter}`,
        transform: 'rotate(-45deg)',
      };
    } else if (i === 7) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: `-${radius}`,
        top: '0',
        transform: 'rotate(90deg)',
      };
    } else if (i === 8) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${lineSpinFade} 1.2s ${delay('0.12s', n, i)} infinite ease-in-out`,
        left: `-${quarter}`,
        top: quarter,
        transform: 'rotate(45deg)',
      };
    }
  }

  return grid;
}

export default (color?: string) => ({
  ...lineSpinFadeLoader(),
  left: '-4px',
  position: 'relative' as 'relative',
  top: '-10px',
  ' > div': {
    ...lines(color),
    ...globalAnimation(),
    height: '15px',
    position: 'absolute' as 'absolute',
    width: '5px',
  },
});
