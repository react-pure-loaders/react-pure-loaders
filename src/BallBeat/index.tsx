/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
