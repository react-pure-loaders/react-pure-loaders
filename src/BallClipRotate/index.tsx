/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
