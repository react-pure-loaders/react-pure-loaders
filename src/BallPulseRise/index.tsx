/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
