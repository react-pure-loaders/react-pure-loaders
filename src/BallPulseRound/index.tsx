/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
