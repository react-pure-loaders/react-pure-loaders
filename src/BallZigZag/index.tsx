/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
