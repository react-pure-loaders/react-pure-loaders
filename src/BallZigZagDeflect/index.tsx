/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
