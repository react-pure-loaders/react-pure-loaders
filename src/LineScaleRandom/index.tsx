/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
