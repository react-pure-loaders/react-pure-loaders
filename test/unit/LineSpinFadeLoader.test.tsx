import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import LineSpinFadeLoader from '../../src/LineSpinFadeLoader';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<LineSpinFadeLoader>', () => {
    afterEach(cleanup);

    test('LineSpinFadeLoader should match snapshot', () => {
        const { container } = render(<LineSpinFadeLoader loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('LineSpinFadeLoader should have default color', () => {
        const { container } = render(<LineSpinFadeLoader loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('LineSpinFadeLoader should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<LineSpinFadeLoader color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('LineSpinFadeLoader should have no children', () => {
        const { container } = render(<LineSpinFadeLoader loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('LineSpinFadeLoader should have three children', () => {
        const { container } = render(<LineSpinFadeLoader loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(9);
    });
});
