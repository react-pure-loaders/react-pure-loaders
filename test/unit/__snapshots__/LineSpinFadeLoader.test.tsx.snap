// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineSpinFadeLoader> LineSpinFadeLoader should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  50% {
    opacity: 0.3;
  }
}

.emotion-0 {
  left: -4px;
  position: relative;
  top: -10px;
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation: animation-0 1.2s -0.84s infinite ease-in-out;
  animation: animation-0 1.2s -0.84s infinite ease-in-out;
  left: 0;
  top: 20px;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation: animation-0 1.2s -0.72s infinite ease-in-out;
  animation: animation-0 1.2s -0.72s infinite ease-in-out;
  left: 13.636363636363637px;
  top: 13.636363636363637px;
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation: animation-0 1.2s -0.60s infinite ease-in-out;
  animation: animation-0 1.2s -0.60s infinite ease-in-out;
  left: 20px;
  top: 0;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-0>div:nth-of-type(4) {
  -webkit-animation: animation-0 1.2s -0.48s infinite ease-in-out;
  animation: animation-0 1.2s -0.48s infinite ease-in-out;
  left: 13.636363636363637px;
  top: -13.636363636363637;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.emotion-0>div:nth-of-type(5) {
  -webkit-animation: animation-0 1.2s -0.36s infinite ease-in-out;
  animation: animation-0 1.2s -0.36s infinite ease-in-out;
  left: 0;
  top: -20px;
}

.emotion-0>div:nth-of-type(6) {
  -webkit-animation: animation-0 1.2s -0.24s infinite ease-in-out;
  animation: animation-0 1.2s -0.24s infinite ease-in-out;
  left: -13.636363636363637;
  top: -13.636363636363637;
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-0>div:nth-of-type(7) {
  -webkit-animation: animation-0 1.2s -0.12s infinite ease-in-out;
  animation: animation-0 1.2s -0.12s infinite ease-in-out;
  left: -20px;
  top: 0;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-0>div:nth-of-type(8) {
  -webkit-animation: animation-0 1.2s 0.00s infinite ease-in-out;
  animation: animation-0 1.2s 0.00s infinite ease-in-out;
  left: -13.636363636363637;
  top: 13.636363636363637px;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.emotion-0>div {
  border-radius: 2px;
  height: 15px;
  margin: 2px;
  width: 5px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  position: absolute;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
